protocol/messenger.go

```diff
@@ -2268,7 +2268,8 @@ func (m *Messenger) dispatchMessage(ctx context.Context, rawMessage common.RawMe
 			)
 			return rawMessage, fmt.Errorf("can't post message type '%d' on chat '%s'", rawMessage.MessageType, chat.ID)
 		}
-
+		//setting content-topic over-ride for community messages to use memberUpdatesChannelID
+		rawMessage.ContentTopicOverride = community.MemberUpdateChannelID()
```

Comment:
The reason chatID was not over-ridden is because the same method `LoadPublic` is used to create a filter that is not present and also load an existing filter. 
In phase-1 we want to send using only universal-content-topic but we want to receive using both universal and chatID based content-topics to have compatibility.
If i override `chatID` then i was not sure if in some flow filter gets created via LoadPublic may get messed up. 
But let me try changing it and see if it works.

Reply:
`HashRatchetGroupID` should be unrelated in this case. I would try go go with `chatID` override, if tests are passing (we have pretty good protocol test coverage), then you can assume it is working as expected. I can't think of any flow that might be broken by this atm.

```golang
// Use a single content-topic for all community chats.
// Reasoning: https://github.com/status-im/status-go/pull/5864
rawMessage.LocalChatID = community.UniversalChatID()
```

Reply:
I [tried it](https://github.com/status-im/status-go/pull/6179) and some tests are failing. I'll investigate why next week..

Reply:
I don't think we can override chatID here. Because https://github.com/status-im/status-go/blob/e49e585349323146a91deac1bd3c1afe12085c23/protocol/transport/transport.go#L287 we need to pass contentTopicOverride to load the filter. 

If we don't pass it there is a chance that wrong filter may get loaded leading to using wrong encryption (as encryption is based on chatID at https://github.com/status-im/status-go/blob/e49e585349323146a91deac1bd3c1afe12085c23/protocol/transport/filters_manager.go#L623-L631. 

This is the reason i had to pass a separate field called overrideContentTopicID so that the original chatID(is used to get the appropriate encryption/decryption key for public chats) and overrideChatID is used to derive the contentTopic as explained [here](https://github.com/status-im/status-go/pull/5864/files#r1825662724).

Reply:
The tests are passing now. Please take a look: https://github.com/status-im/status-go/pull/6179.

> If we don't pass it there is a chance that wrong filter may get loaded leading to using wrong encryption (as encryption is based on chatID as can be seen below)

Indeed, symmetric key generation occurs at the filter level, with the key derived from the `chatID`, aka `contentTopic`. It is the responsibility of the higher layer to initialize all filters properly, and as far as I can see, it is done, as filters for both the `UniversalChatID` and each `ChannelID` (to ensure backward compatibility) are being registered.

Reply:
Got it, I will test with this PR and older version to make sure there are no issues.
This definitely looks cleaner than mine.